packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -962,6 +962,10 @@ export class BIGINT extends BaseIntegerDataType {
   protected _checkOptionSupport(dialect: AbstractDialect) {
     super._checkOptionSupport(dialect);
 
+    if (!dialect.supports.dataTypes.BIGINT) {
```

Comment:
I don't understand why this change is necessary. We currently have bigint support and the value is correctly returned as a string for sqlite in the existing data-types.test.ts

Reply:
While BIGINT is currently supported, its support is not always valid.

For instance, this test is currently skipped as the value returned by the DB is a JS number which obviously causes values greater than the JS number supports will be invalid.

https://github.com/sequelize/sequelize/blob/c351a9057122df30886bcad2a8a558759c5374a8/packages/core/test/integration/data-types/data-types.test.ts#L630-L640

This test has to have a condition for SQLite there is no way to parse the database value and convert it to a string in its raw form. The way for the current implementation to return string for the BIGINT data type is for the value to have its data type specified as BIGINT which by default converts the value to string as per below.

https://github.com/sequelize/sequelize/blob/c351a9057122df30886bcad2a8a558759c5374a8/packages/core/test/integration/data-types/data-types.test.ts#L658-L669

By default, `parseDatabaseValue` calls `sanitize` which in the case of BIGINT, returns the data base value as string.

https://github.com/sequelize/sequelize/blob/c351a9057122df30886bcad2a8a558759c5374a8/packages/core/src/dialects/abstract/data-types.ts#L970-L982

Reply:
Oh yep, right
Disabling bigints in sqlite is OK then